runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -739,9 +741,12 @@ public final void processWatermark1(Watermark mark) throws Exception {
     }
 
     currentInputWatermark = mark.getTimestamp();
```

Comment:
We need the following to generalize watermark emission to be able to call this method from other places. 

```
if (mark.getTimestamp > currentInputWatermark) {
  currentInputWatermark = mark.getTimestamp();
}

Reply:
Input watermark is not what is concerned by this PR. That logic did not change, we only need to be able to progress output watermark when watermark hold changes.

Reply:
You would need this change in order to go through the regular watermark emission code without changing the latest seen input watermark.

Reply:
I would like to generalize the existing code, instead of adding code paths for every exception which is bound to be error-prone.